src/metabase/db.clj

```diff
@@ -254,6 +254,20 @@
       (catch Exception e
         (log/error e (trs "Unable to release the Liquibase lock after a migration failure"))))))
 
+(defn- print-migrations-and-quit-if-needed!
+  "If we are not doing auto migrations and there are unrun migrations then print out migration SQL for user to
+  run manually. Then throw an exception to short circuit the setup process and make it clear we can't proceed."
+  [liquibase]
+  (when (has-unrun-migrations? liquibase)
+    (println (let [sql (migrations-sql liquibase)]
```

Comment:
I think this `println` is not correct. This was wrong in the code before as well. The SQL is being written via the log message below `log/info`. The result of that invocation will be nil, then an exception will be thrown.

Usually we always want to log things and not use `println`. In this case though, the user is starting up Metabase to see the migrations and wants that written to the console. It seems like we should swap the `log/info` for a `println`. Then we can remove this outer `println`